compiler/src/iree/compiler/Codegen/Common/test/propagate_reshapes_by_expansion.mlir

```diff
@@ -86,3 +87,103 @@ func.func @fold_collapse_into_stores_dynamic(%arg0 : tensor<2x?x32xf32>) {
 //       CHECK:   flow.dispatch.tensor.store %{{.+}}, %[[SUBSPAN]]
 //  CHECK-SAME:       offsets = [0, 0, 0], sizes = [2, %[[SHAPE]], 32], strides = [1, 1, 1]
 //  CHECK-SAME:       !flow.dispatch.tensor<writeonly:tensor<2x?x32xf32>>{%[[SHAPE]]}
+
+// -----
```

Comment:
Can you also add a test with dynamic dims? Maybe one with a dynamic dim blocking expansion, and one with mixed dynamic and static dims that can be expanded?

Maybe also something that tests the conditions in `verifyAndCollectExpandableUsers`.

Reply:
Can definitely add the test but will have to see if its easy to actually do the expansion. Usually in dynamic case there is an extract_slice after the collapse so this pattern cant be applied anyway so I dont see much value in supporting the dynamic case.

Reply:
I see, yeah I think it makes sense for packed dynamic dimensions. I was also imagining a case like a batch matmul with a dynamic batch dimension. The static M/N/K dimensions would be unpacked/collapsed, but the outer batch dimension is untouched. I think we want that to still work for this pattern right?

Reply:
Ya that makes sense, so we basically want to support dynamic destination sizes then, is that right?